utils/url.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTILS_URL_H
+#define UTILS_URL_H
+
+#include "defined_error.h"
+
+/**
+ * @brief send message via http protocol
+ *
+ * @param[in] host https host
+ * @param[in] port https port
+ * @param[in] api https api for post request
```

Comment:
The type of `api` is `char *`, which might not reflect what it behave as "https api." You shall clarify.

Reply:
Also, according to the usage, this file should be renamed to `utils/protocol.h` rather than `utils/url.h`.

Reply:
Is it better to concatenate host and api?

Reply:
It is indeed referring to host, rather than generic api.